README.md

```diff
@@ -9,6 +9,34 @@
 npm install screwdriver-template-main
 ```
 
+### Publishing
+
+To publish a new template or an updated version of a template, include `template: screwdriver/publishTemplate` in your job. This installs the `template-main` npm package to the build container that your build is running in, so your build will be able to access `publish` script.
+
+A template maintainer would specify their template in a `yaml`. By default, the path `./sd-template.yaml` will be read. However, a user can specify a custom path using the env variable: `SD_TEMPLATE_PATH`.
+
+To publish multiple templates in the same repo, a template maintainer would use the following pattern in their `screwdriver.yaml`:
+
+```
+...
+
+jobs:
+    main:
+        steps:
+            - random: echo 'random step'
+    publishfirst:
+        template: screwdriver/publishTemplate
+        environment:
+            SD_TEMPLATE_PATH: ./path/to/my-template.yaml
+    publishsecond:
```

Comment:
Also, we need to make sure that the publish endpoint validates the template. We don't want the user to be able to upload invalid templates.